File: cs50x/cash/cash.c
#include <cs50.h>
#include <stdio.h>

int get_change_owed(void);
int calculate_num_coins(int cash_owed);

int main(void)
{
    int change_owed = get_change_owed();
    int num_coins = calculate_num_coins(change_owed);

    printf("%i\n", num_coins);
}

int get_change_owed(void)
{
    int change_owed;

    // ask the user for the amount of change owed
    // only accept a positive integer
    do
    {
        change_owed = get_int("Change owed: ");
    }
    while (change_owed < 0);

    return change_owed;
}

int calculate_num_coins(int cash_owed)
{

    // calculate the number of quarters in cash_owed
    int quarters = cash_owed / 25;
    // remove the quarters in cash_owed
    cash_owed -= (quarters * 25);

    // calculate the number of dimes left in cash_owed
    int dimes = cash_owed / 10;
    // remove the dimes in cash_owed
    cash_owed -= (dimes * 10);

    // calculate the number of nickles left in cash_owed
    int nickles = cash_owed / 5;
    // remove the nickles in cash_owed
    cash_owed -= (nickles * 5);

    // calculate the number of pennies left in cash_owed
    int pennies = cash_owed / 1;
    // remove the pennies in cash_owed
    cash_owed -= (pennies * 1);

    // add the coins
    int num_coins = quarters + dimes + nickles + pennies;

    return num_coins;
}

File: cs50x/sentimental-cash/cash.py
from cs50 import get_float

DOLLAR = 100
QUARTER = 25
DIME = 10
NICKEL = 5
PENNY = 1

# Function to prompt for change owed


def prompt_for_change_owed():
    while True:
        change_owed = get_float("Changed owed: ")

        # Check to see if input is valid
        if change_owed > 0:
            return change_owed


# Function to prompt for the number of coins in the change owed
def calculate_coins_needed(change_in_cents):

    # Treat coins as integers, convert dollars and cents
    change_in_cents *= DOLLAR

    coins = 0

    # Do this until there is no more change owed
    # If change_owed > 1, we can still take coins from it
    while change_in_cents >= PENNY:

        # Count quarters
        quarters = int(change_in_cents / QUARTER)

        # Remove quarters from change owed
        change_in_cents -= QUARTER * quarters

        # Count dimes
        dimes = int(change_in_cents / DIME)

        # Remove dimes from change owed
        change_in_cents -= DIME * dimes

        # Count nickels
        nickels = int((change_in_cents / NICKEL))

        # Remove nickles from change owed
        change_in_cents -= NICKEL * nickels

        # Count pennies
        pennies = int((change_in_cents / PENNY))

        # Remove pennies from change owed
        change_in_cents -= PENNY * pennies

    coins = quarters + dimes + nickels + pennies

    return coins


# Print the number of coins in the change owed
def display_coins(coins):
    print(f"{coins}")


# Main function calls the working functions
def main():
    # Get input for change owed
    change_owed = prompt_for_change_owed()

    # Call function to calculte how many coins of what denomination
    coins = calculate_coins_needed(change_owed)

    # Calls the function to print the results
    display_coins(coins)


# Calls the main function to start the program
main()
